### PR TITLE
Use interpolated strings 12

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replace `string.Format()` with new `string.Create((IFormatProvider? provider, [InterpolatedStringHandlerArgument(nameof(provider))] ref DefaultInterpolatedStringHandler handler)`.

## PR Context
#18974, I think this is the last one

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by CarloToso** [comment]
@iSazonov What is the preferred solution ? `streamWriter.NewLine` -> `streamWriter.NewLine.ToString()` or `streamWriter.NewLine` -> `Environment.NewLine`

**Comment by iSazonov** [comment]
> @iSazonov What is the preferred solution ? `streamWriter.NewLine` -> `streamWriter.NewLine.ToString()` or `streamWriter.NewLine` -> `Environment.NewLine`

We can not use `Environment.NewLine` since streamWriter can use any value. 
`streamWriter.NewLine` doesn't work but `streamWriter.NewLine.ToString()`?  If so we can use second one.

**Comment by iSazonov** [comment]
Maybe it's not worth the effort.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 1]
```diff
@@ -886,14 +886,14 @@ private List<string> TryResolveFilePath(string filePath)
         /// <returns></returns>
         private string ManifestFragment(string key, string resourceString, string value, StreamWriter streamWriter)
         {
-            return string.Format(CultureInfo.InvariantCulture, "{0}# {1}{2}{0}{3:19} = {4}{2}{2}",
-                _indent, resourceString, streamWriter.NewLine, key, value);
+            string nl = streamWriter.NewLine;
+            return string.Format(CultureInfo.InvariantCulture, "{0}# {1}{2}{0}{3:19} = {4}{2}{2}", _indent, resourceString, nl, key, value);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs:890]: I think it makes no sense to change something in the file. Please revert.

File: src/System.Management.Automation/engine/remoting/commands/NewPSSessionConfigurationFile.cs [hunk 2]
```diff
@@ -1866,10 +1866,10 @@ internal static string ConfigFragment(string key, string resourceString, string
 
             if (isExample)
             {
-                return string.Format(CultureInfo.InvariantCulture, "# {0}{1}# {2:19} = {3}{4}{5}", resourceString, nl, key, value, nl, nl);
+                return string.Format(CultureInfo.InvariantCulture, "# {0}{1}# {2:19} = {3}{1}{1}", resourceString, nl, key, value);
             }
-            
-            return string.Format(CultureInfo.InvariantCulture, "# {0}{1}{2:19} = {3}{4}{5}", resourceString, nl, key, value, nl, nl);
+
+            return string.Format(CultureInfo.InvariantCulture, "# {0}{1}{2:19} = {3}{1}{1}", resourceString, nl, key, value);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/commands/NewPSSessionConfigurationFile.cs:1872]: The same.

**Review by iSazonov — APPROVED** [review]

**Review by iSazonov — CHANGES_REQUESTED** [review]
@CarloToso Again an test fails.

**Comment by CarloToso** [comment]
@iSazonov I don't know why the tests are failing, it doesn't make any sense

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/runtime/Binding/Binders.cs [hunk 3]
```diff
@@ -3959,12 +3959,7 @@ private PSGetIndexBinder(Tuple<CallInfo, PSMethodInvocationConstraints, bool> tu
 
         public override string ToString()
         {
-            return string.Format(CultureInfo.InvariantCulture,
-                                 "PSGetIndexBinder indexCount={0}{1}{2} ver:{3}",
-                                 this.CallInfo.ArgumentCount,
-                                 _allowSlicing ? string.Empty : " slicing disallowed",
-                                 _constraints == null ? string.Empty : " constraints: " + _constraints,
-                                 _version);
+            return string.Create(CultureInfo.InvariantCulture, $"PSGetIndexBinder indexCount={this.CallInfo.ArgumentCount}{(_allowSlicing ? string.Empty : " slicing disallowed")}{(_constraints == null ? string.Empty : " constraints: " + _constraints)} ver:{_version}");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/runtime/Binding/Binders.cs:3962]: @iSazonov I think this is way less readable than the original code, don't you think so?

Comment (iSazonov) [reply src/System.Management.Automation/engine/runtime/Binding/Binders.cs:3962]: I think more about reducing allocations. My suggestions for improvement I outlined in your PR.

**Pull request merged**